tests/typecheck/fields/test_custom_fields.yml

```diff
@@ -0,0 +1,62 @@
+-   case: test_custom_model_fields_with_generic_type
+    main: |
+        from myapp.models import User, CustomFieldValue
+        user = User()
+        reveal_type(user.id)  # N: Revealed type is "builtins.int"
+        reveal_type(user.my_custom_field1)  # N: Revealed type is "myapp.models.CustomFieldValue"
+        reveal_type(user.my_custom_field2)  # N: Revealed type is "myapp.models.CustomFieldValue"
+        reveal_type(user.my_custom_field3)  # N: Revealed type is "builtins.bool"
+        reveal_type(user.my_custom_field4)  # N: Revealed type is "myapp.models.CustomFieldValue"
+        reveal_type(user.my_custom_field5)  # N: Revealed type is "Union[myapp.models.CustomFieldValue, None]"
+        # user.my_custom_field6 is incorrectly typed as non-optional
+        # reveal_type(user.my_custom_field6) ## N: Revealed type is "Union[myapp.models.CustomFieldValue, None]"
+        reveal_type(user.my_custom_field7)  # N: Revealed type is "Union[builtins.bool, None]"
+        reveal_type(user.my_custom_field8)  # N: Revealed type is "Union[myapp.models.CustomFieldValue, None]"
+        reveal_type(user.my_custom_field9)  # N: Revealed type is "myapp.models.CustomFieldValue"
+        reveal_type(user.my_custom_field10)  # N: Revealed type is "Union[myapp.models.CustomFieldValue, None]"
```

Comment:
I would say the behaviour for fields: `my_custom_field5`, `my_custom_field8` and `my_custom_field10` is ambiguous. Since e.g. `my_custom_field5` is explicitly annotated like:

`GenericField[Union[CustomFieldValue, int], CustomFieldValue](null=True)`

But the get type is then rewritten from `CustomFieldValue` to `Union[myapp.models.CustomFieldValue, None]` due to `null=True`. I get the convenient part of this, but when passing generic types explicitly I'd argue we shouldn't rewrite. Especially not without any note or message.

In some sense I think the explicitly passed get type(`CustomFieldValue`) and the underlying genericness of a field, triggered by `null=True`, could raise an incompatibility error. I wouldn't find it strange if the declaration of `my_custom_field5`, `my_custom_field8` or `my_custom_field10` yielded an error regarding the mentioned incompatibility, saying that their _get_ type needs include `None` as an additional type.

I have no idea if the above is related to this PR or not though.

Reply:
I will also remove the code that rewrites fields with null=True to be optional and we'll rely on developers being explicit

Reply:
Sounds good 👍